### ISSUE TITLE
[TW#22512] Directed Advertisement only reported once per device from BT Controller

### ISSUE THREAD

**Issue by jakkra** [espressif/esp-idf#1933]
INSTRUCTIONS
============

Before submitting a new issue, please follow the checklist and try to find the answer.

- [x] I have read the documentation [ESP-IDF Programming Guide](http://esp-idf.readthedocs.io/en/latest/) and the issue is not addressed there.
- [ ] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [x] I have searched the issue tracker for a similar issue and not found a similar issue.

If the issue cannot be solved after the steps before, please follow these instructions so we can get the needed information to help you in a quick and effective fashion.

## Environment
- IDF version: 94ec3c8e53e7f48133f17ae6c6e905fa5a856fd2
- Development Env:		[Make]
- Operating System:		[Windows]
- Power Supply:         [USB]

## Problem Description

If I send a directed advertisement to a ESP32, it will only be reported once for each device sending a directed advertisement. If a directed advertisement is sent to a ESP32 Central and this central choose to not initiate a connection, any further directed advertisements from that device will NOT be passed to the application, even after waiting minutes or after connection changes. Example to easier explain the issue:

Setup:
- Device A Central ESP32
- Device B Peripheral ESP32
- Device C Peripheral ESP32

1. Device A initiates a connection to B => A and B are now connected.
2. Device C Sends a directed advertisement to A ->directed adv. reported from BT Controller to application -> application ignores it, because it can't handle a new connection right now.
3. Device C retries after 30s and sends new directed adv. to A -> Never reported from BT controller to application.
4. A sends disconnect to B => A and B are no longer connected. 
5. Device C retries yet again to initiate a connection to A using a directed adv. -> Directed adv. never reported to the application. 

Additional info:
If I in any step send a directed advertisement from an other device X, it will be reported to the application in device A. Meaning I'm sure device A is scanning in the background.

Restarting scanning does generate a new directed advertisement.

This problem is likely in the BT Controller, since I can see on HCI level that the directed advertisement is only reported once for each device. We are using the Bluekitchen stack, and I have attached the package 
HCI log received from the BT Controller doing the steps explained above. 

### Expected Behavior
- The directed advertisement should be reported to the application in device A in step 3.
- The directed advertisement shall be reported to the application in device A in step 5.
### Actual Behavior
No directed advertisement reported to the application in steps 3 and 5.
### Steps to repropduce

Explained in Problem Description

## Debug Logs

I have added the raw HCI LOG and a .pklg which can be opened in Wireshark or similar tool.
There are alot of advertisements, but you can filter on directed ones using filter for example:
bthci_evt.le_advts_event_type == 0x01

[dump.zip](https://github.com/espressif/esp-idf/files/1979051/dump.zip)

**Comment by Weijian-Espressif** [comment]
@jakkra  When Controller received directed adv, it will  automatically enable scan duplicate , so the directed advertisement is only reported to host  only once for each device.

**Comment by jakkra** [comment]
@Weijian-Espressif Thank you, I suspected that was the case. What I did was to always restart the passive scan if I couldn't handle the directed advertisement.

You may close this issue.
